### DISCUSSION TITLE
Quartz .net call scheduler outside of DI container or Main method

### DISCUSSION THREAD

**Discussion by MaxBrooks114** [quartznet/quartznet#1698] [Q&A]
We are trying to use quartz for some of our scheduled tasks such as email and record duplication. It seems to be a perfect fit however,  we have an issue. We have a separate project in our solution where we want to house our scheduler and different jobs, which will have access to our service layer where each Job calls our services. The issue is, we aren't sure how to get access to the scheduler after it gets instantiated in the job runner project's main file.

Main file:

  private static async Task Main(string[] args)
 {
   LogProvider.SetCurrentLogProvider(new ConsoleLogProvider());
   // Grab the Scheduler instance from the Factory
   StdSchedulerFactory factory = new StdSchedulerFactory();
   IScheduler scheduler = await factory.GetScheduler();
   
// and start it off
   await scheduler.Start();
   
// and last shut down the scheduler when you are ready to close your program
   await scheduler.Shutdown();
}
Typical Job:

using Quartz;
using Compyl.AppLogic.AssessmentService;

namespace JobRunner.Jobs
{
    public class DuplicationJob : IJob
    {
        private readonly IService _Service;
       
        public DuplicationJob(IService Service)
        {
            _service = Service;
          
        }

        public async Task Execute(IJobExecutionContext context)
        {
            ....
        }
    }
}

we want to have a class or method in the job runner project that houses the scheduler through DI and a method(s) like

[Inject] IScheduler scheduler
public ScheduleEntity (T entity)
{
   var job = createJob(entity)//another method that creates the job and jobdatamap this is easy enough
   var trigger = createTrigger(entity)// same as above
   scheduler.ScheduleJob(job, trigger)

}

Then we would call the ScheduleEntity method wherever we needed it in our webapp.

**Comment by lahma** [comment]
https://stackoverflow.com/questions/73284259/quartz-net-call-scheduler-outside-of-di-container-or-main-method

**Reply by MaxBrooks114** [reply]
That's my post. The documentation is still extremely unclear as to what we should inject and how we should inject it. Why are jobs/ triggers being scheduled in the startup of the app? How do we schedule jobs outside of the DI container??
